### PR TITLE
configure.ac: cleanup device configure checks

### PR DESCRIPTION
**Pull request by aingerson**
Cleanup cuda, rocr, gdrcopy, and neuron config checks to minimize DL dependencies, make consistent defines, and make more readable.

### CODE REVIEW

**Comment by aingerson** [comment]
@wzamazon @iziemba Would appreciate a review from anyone who has hardware to verify these enable paths!

**Comment by shefty** [comment]
It looks like the psm3 provider breaks because it's own configure scripts look for have_libcuda, which was removed.  I can't easily tell what it actually needs - just the headers or the library itself.

**Comment by aingerson** [comment]
Fixed psm3 issue and some missed ifdef removals

**Comment by j-xiong** [comment]
Jenkins CI error message:
```
script.sh: line 4: cd: /opt/jenkinsagent/workspace/ofi_libfabric_PR-7467/contrib/intel/jenkins/: No such file or directory
```

**Comment by shefty** [comment]
I restarted the CI.  Not sure why that error occurred.

File: prov/psm3/configure.m4 [hunk 1]
```diff
@@ -138,6 +138,8 @@ ifelse('
 			])
 		CFLAGS=$save_CFLAGS
 
+		have_libcuda=0
+		AS_IF([test $have_cuda -eq 1 && test $cuda_dlopen -eq 0], [have_libcuda=1])
```
Comment (acgoldma) [inline prov/psm3/configure.m4:142]: psm3 required dlopen to use cuda currently.

Comment (aingerson) [reply prov/psm3/configure.m4]: Ah I interpreted the check backwards thinking psm3 could only statically link. Thanks for catching. I changed the check to make sure enable_cuda_dlopen is set to 1

**Review by acgoldma — APPROVED** [review]
PSM3 changes looks okay. Thanks.

**Pull request merged**